### DISCUSSION TITLE
Python.exe keeps crashing V 3.10.6 - Win11, x64

### DISCUSSION THREAD

**Discussion by zejacky** [deforum-art/sd-webui-text2video#17] [Q&A] [answered]
First I want to say thank you very much for this amazing extension.
I was able to do already my first animation of an old man running:-)

However, I face many python.exe crashes
.
I see the following message inside A1111:
Starting text2video
False
DDIM sampling tensor(463): ca 55%
after a while "Press any key to continue . . . "

System:
Windows 11, x64
Intel PC with RTX 4090 24GB VRAM and 64 GB RAM.

Event log error message:
Faulting application name: python.exe, version: 3.10.6150.1013, time stamp: 0x62e84c21
Faulting module name: python310.dll, version: 3.10.6150.1013, time stamp: 0x62e84bd6
Exception code: 0xc0000005
Fault offset: 0x0000000000002e31
Faulting process ID: 0x0x2B68
Faulting application start time: 0x0x1D95B44EDDF6075

I had the same errors with ControlNet. Sometimes it works perfect, other times one crash followed by another.
( pip version 22.2.1)

What I already did:
- Installing python 3.10.9 and 3.10.10
- RAM Test with memtest86, 48 passes
- Uninstalling Adguard
- Uninstalling many other programs
- Git clone new A1111
- Installing only this extension
- Created the folder models\ModelScope\t2v and put the content from damo-vilab/modelscope-damo-text-to-video-synthesis inside

Additional Infos:
-  Launching Web UI with arguments: --xformers --autolaunch --theme dark
- ..\venv\lib\site-packages\imageio_ffmpeg\binaries\ffmpeg-win64-v4.2.2.exe

I hope someone have an idea, why python is (randomly?) crashing. 
Sometimes I can do a movie, and after 10 crashes maybe another.

Did I miss something?

Thank you very much!

**Comment by kabachuha** [comment, marked as answer]
It's really strange, especially if you can generate a plenty of frames quite well.

My two guesses are:

1. It may be caused by xformers breaking stuff on your setup.
2. There is VAE crash happening somewhere during the generation process, caused by the lack of precision your GPU cannot support (so only fp32, not half fp16). Model halving is hardcoded at the moment, but it's proper handling (esp related to the VAE) is on the list.

**Reply by zejacky** [reply]
> It's really strange, especially if you can generate a plenty of frames quite well.
> 
> My two guesses are:
> 
> 1. It may be caused by xformers breaking stuff on your setup.
> 2. There is VAE crash happening somewhere during the generation process, caused by the lack of precision your GPU cannot support (so only fp32, not half fp16). Model halving is hardcoded at the moment, but it's proper handling (esp related to the VAE) is on the list.

Thank you for your answer!
I use following config:  python: 3.10.9  •  torch: 1.13.1+cu117  •  xformers: 0.0.16rc425  •  gradio: 3.16.2  •  commit: a9fed7c3

After the new update of "ModelScope text2video extension for auto1111" (ab1c4e74), I have no more crashes!
Running with booth GPU (half precision) / GPU
(Need more testing)

Could you please explain the possible xformers breaking stuff ? Maybe I could use the knowledge for other errors.

**Reply by kabachuha** [reply]
Glad it works now 🙂

There were some determinism problems caused by xformers, so I thought it might cause sporadic problems here as well
